src/main/kotlin/com/hypto/iam/server/service/UsersService.kt

```diff
@@ -538,6 +554,136 @@ class UsersServiceImpl : KoinComponent, UsersService {
         val userHrn = ResourceHrn(userRecord.organizationId, "", IamResources.USER, user.username)
         return getUser(userHrn, userRecord)
     }
+
+    override suspend fun linkUser(
+        principal: UserPrincipal,
+        request: LinkUserRequest,
+    ): TokenResponse {
+        val inviteeHrn =
+            when (request.type) {
+                LinkUserRequest.Type.EMAIL_APPROVAL -> throw BadRequestException("WILL BE IMPLEMENTED IN USER LINKS V2")
+                LinkUserRequest.Type.USERNAME_PASSWORD -> {
+                    if (!appConfig.app.uniqueUsersAcrossOrganizations) {
+                        throw BadRequestException("Email not unique across organizations. Please use link user passcode")
```

Comment:
Everything is implemented when `uniqueUsersAcrossOrganizations` is true.
Should we ask for nullable organizationId, if provided and `uniqueUsersAcrossOrganizations` is false, then everything can work without issues